### PR TITLE
Instant Search: add more styling specificity to search input

### PR DESCRIPTION
**Pull request by keoshi**
Fixes https://github.com/Automattic/jetpack/issues/14608

#### Changes proposed in this Pull Request:
* Ensure there's a cross to clear search on browsers that support it.
* Reduce font-size on results count text.
* Make close button behave similarly on different themes.

image

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* No.

#### Testing instructions:

1. Follow [these setup instructions](https://github.com/Automattic/jetpack/blob/instant-search-master/modules/search/instant-search/README.md#testing-instructions) to set up Jetpack Instant Search on your site.
2. Search your site to trigger the Jetpack Search overlay.
3. Ensure that the Jetpack Search input field, the result count text, and the close button are styled correctly on multiple themes.

#### Proposed changelog entry for your changes:
* None.

### CODE REVIEW

**Comment by keoshi** [comment]
@jsnmoon @gibrown Tried to rebase my PR to get the latest search input changes on `instant-search-master` and validate if they are compatible, but I don't think I've done it correctly. Any help would be appreciated. Just want to ensure the changes on this PR along with https://github.com/Automattic/jetpack/pull/14629 are enough to close the first bullet point of #14608:

> control the size and the box itself rather than inheriting it all.

**Comment by keoshi** [comment]
Quick follow up @gibrown @jsnmoon : created a complementary PR to this one on https://github.com/Automattic/jetpack/pull/14664

**Review by gibrown — APPROVED** [review]
LGTM. Excited to see this combined with the other PRs

**Review by jsnmoon — COMMENTED** [review]
I'm not seeing anything amiss on with the rebase. 

However, I'm not able to see the cross to clear the search input on the latest Chrome 80, Firefox 73, and Safari 13 in macOS. Is this expected?

If so, feel free to 🚢. Otherwise, further attention may be warranted.

**Comment by keoshi** [comment]
@jsnmoon Sorry, I should have specified it more clearly. Firefox does not support it, but Chrome and Safari should display the cross as it's their default behavior. Let's address that if you're not seeing it.

**Comment by jeffgolenski** [comment]
@keoshi Visually, I'm worried that the [x] icon isn't visible enough. It's fairly light gray and small. Have we considered using an X without the circle so it's larger?

**Review by jeffgolenski — CHANGES_REQUESTED** [review]
See above comment

**Comment by jsnmoon** [comment]
> Chrome and Safari should display the cross as it's their default behavior @keoshi 

Unfortunately, this is what I see on Chrome 80:
Screen Shot 2020-02-14 at 3 43 30 PM

Not faring much better on Safari 13.0.5:
Screen Shot 2020-02-14 at 3 43 48 PM

**Review by jsnmoon — APPROVED** [review]

**Comment by keoshi** [comment]
> Visually, I'm worried that the [x] icon isn't visible enough. It's fairly light gray and small. Have we considered using an X without the circle so it's larger?

@jeffgolenski What you're seeing is the browser's implementation of a built-in feature. We could add our own, but I'd like to avoid removing the default one and rebuilding it.

**Comment by keoshi** [comment]
@jsnmoon Tried adding more specificity but couldn't make it happen for 2020. Any idea what else we can do here?

**Comment by jeffgolenski** [comment]
@keoshi no problem - carry on. I love the idea of using what's already built in!

File: modules/search/instant-search/components/search-box.scss [hunk 1]
```diff
@@ -50,4 +50,11 @@ input.jetpack-instant-search__box-input.search-field {
 .jetpack-instant-search__box input,
 .jetpack-instant-search__box input[type='search'].jetpack-instant-search__box-input {
 	width: 100%;
+
+	&::-webkit-search-decoration,
+	&::-webkit-search-cancel-button,
+	&::-webkit-search-results-button,
+	&::-webkit-search-results-decoration {
+		-webkit-appearance: searchfield;
```
Comment (jsnmoon) [inline modules/search/instant-search/components/search-box.scss]:
```suggestion
		-webkit-appearance: searchfield;
		display: initial;
```

This should fix the issue with 2020. These elements were being set to `display: none`.

Comment (keoshi) [reply modules/search/instant-search/components/search-box.scss]: Brilliant @jsnmoon ! Thanks for figuring that out. Added in https://github.com/Automattic/jetpack/pull/14661/commits/9d6d6b2920cecbcc061cf28e072ac7446c41fe2b

**Comment by keoshi** [comment]
Resolved conflicts and merging now since this has already been tested despite what the labels say. :)

**Pull request merged**